dali/pipeline/operators/reader/loader/video_loader.h

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2017, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_READER_LOADER_VIDEO_LOADER_H_
+#define DALI_PIPELINE_OPERATORS_READER_LOADER_VIDEO_LOADER_H_
+
+extern "C" {
+#include <libavformat/avformat.h>
+#include <libavcodec/avcodec.h>
+}
+
+#include <thread>
+#include <algorithm>
+#include <random>
+
+#include "dali/common.h"
+#include "dali/pipeline/operators/reader/loader/loader.h"
+#include "dali/pipeline/operators/reader/nvdecoder/nvdecoder.h"
+#include "dali/pipeline/operators/reader/nvdecoder/sequencewrapper.h"
+#include "dali/pipeline/operators/reader/nvdecoder/nvcuvid.h"
+
+//namespace {
+
+// libav resource free function take the address of a pointer...
+template<typename T>
+class AVDeleter {
+  public:
+    AVDeleter() : deleter_(nullptr) {}
```

Comment:
I would remove this default constructor. If someone use it `operator()` will crash.